File: 240124/흰검 칠하기/painting-white-black.py
n = int(input())

color =["FALSE"]*2001

tile = [0] * 2001

right,left,now = 0 ,0 ,1000
for _ in range(n):
    toilet = input().split()

    if toilet[1] =='R':
        right = now + int(toilet[0])
        left = now
        now = right

        for i in range(left,right):
            tile[i]+=1
            color[i]='Black'

    elif toilet[1] =='L':
        right = now 
        left = now - int(toilet[0])
        now = left

        for i in range(left,right):
            tile[i]+=1
            color[i]='White'
wh = 0
bl = 0
gr = 0
for i in range(2001):
    if tile[i]>0 and tile[i]<4 and color[i]=='White':
        wh+=1 
    elif tile[i]>0 and tile[i]<4 and color[i]=='Black':
        bl+=1 
    elif tile[i]>3:
        gr+=1
print(wh,bl,gr)

File: 240207/흰검 칠하기/painting-white-black.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int [] tile = new int[201];
        String[] visited = new String[201];
        
        for(int i = 0; i<201; i++){
            visited[i]="No";
        }
        int left = 100;
        int right = 100;
        int now = 100;

        for (int i = 0; i<n; i++){
            int a = sc.nextInt();
            char word = sc.next().charAt(0);
            if (word=='R'){
                right = now + a;
                for (int j = now; j<right; j++){
                    visited[j] ="Black";
                    tile[j] +=1;
                }
                now = right ;
            }
            if (word=='L'){
                left = now - a;
                for (int j = left; j<now; j++){
                    visited[j] ="White";
                    tile[j] +=1;
                }
                now = left ;
            }
            
            
        }
        int white = 0;
        int black = 0;
        int gray = 0 ;
        for (int i = 0; i<201; i++){
            if(visited[i].equals("White")&& tile[i]<4){
                white+=1;
            }
            if (visited[i].equals("Black") && tile[i]<4){
                black+=1;
            }
            if(tile[i]>=4){
                gray+=1;
            }
        }
        System.out.print(white+" "+black+" "+gray);
    }
}